clients/venice-push-job/src/main/java/com/linkedin/venice/spark/input/pubsub/PartitionSplitters.java

```diff
@@ -0,0 +1,151 @@
+package com.linkedin.venice.spark.input.pubsub;
+
+import com.linkedin.venice.spark.input.pubsub.table.VenicePubsubInputPartition;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class PartitionSplitters {
+  // need a method called fullPartitionSplitter, takes in list of partition start and end offsets
+  // and returns a list of VenicePubsubInputPartition splits
+  public static Map<Integer, List<List<Long>>> fullPartitionSplitter(Map<Integer, List<Long>> partitionOffsetsMap) {
```

Comment:
It looks like theres an assumption that the list in entries for Map<Integer, List<Long>> partitionOffsetsMap need to be two entries, a start and an end.  I'm not sure I see yet in this PR how this method is used, but it looks like between this and assembleSegment, an interface which uses a Pair might make more sense?

Or is there a flexibility gain in using a List that I'm missing?